packages/affiliates/libs/uniswap/processor.ts

```diff
@@ -0,0 +1,249 @@
+// its mainly documenting all events based on uniswap typechain build output.
+import { Positions, NftPositions, Pools, Position } from "./models";
+import { exists, convertValuesToString } from "./utils";
+import assert from "assert";
+import { BigNumberish } from "ethers";
+
+type Event = {
+  event: string;
+  args: any[];
+};
+
+type PoolsType = ReturnType<typeof Pools>;
+
+export function NftEvents({ positions }: { positions: ReturnType<typeof NftPositions> }) {
+  const handlers: { [key: string]: (...args: any[]) => Promise<void> } = {
+    async Approval(owner: string | null, approved: string | null, tokenId: BigNumberish | null) {
+      // nothing
+    },
+    async ApprovalForAll(owner: string | null, operator: string | null, approved: null) {
+      // nothing
+    },
+    async Collect(
+      tokenId: BigNumberish | null,
+      recipient: string | null,
+      amount0: BigNumberish | null,
+      amount1: BigNumberish | null
+    ) {
+      // nothing
+    },
+    async DecreaseLiquidity(
+      tokenId: BigNumberish | null,
+      liquidity: BigNumberish | null,
+      amount0: BigNumberish | null,
+      amount1: BigNumberish | null
+    ) {
+      assert(exists(tokenId), "requires token id");
+      assert(exists(liquidity), "requires liquidity");
+      assert(exists(amount0), "requires amount0");
+      assert(exists(amount1), "requires amount1");
+      if (!(await positions.has(tokenId.toString()))) {
+        await positions.create({
+          tokenId: tokenId.toString(),
+          liquidity: liquidity.toString(),
+          amount0: amount0.toString(),
+          amount1: amount1.toString()
+        });
+      } else {
+        await positions.update(tokenId.toString(), {
+          liquidity: liquidity.toString(),
+          amount0: amount0.toString(),
+          amount1: amount1.toString()
+        });
+      }
+    },
+    async IncreaseLiquidity(
+      tokenId: BigNumberish | null,
+      liquidity: BigNumberish | null,
+      amount0: BigNumberish | null,
+      amount1: BigNumberish | null
+    ) {
+      assert(exists(tokenId), "requires token id");
+      assert(exists(liquidity), "requires liquidity");
+      assert(exists(amount0), "requires amount0");
+      assert(exists(amount1), "requires amount1");
+
+      if (!(await positions.has(tokenId.toString()))) {
+        await positions.create({
+          tokenId: tokenId.toString(),
+          liquidity: liquidity.toString(),
+          amount0: amount0.toString(),
+          amount1: amount1.toString()
+        });
+      } else {
+        await positions.update(tokenId.toString(), {
+          liquidity: liquidity.toString(),
+          amount0: amount0.toString(),
+          amount1: amount1.toString()
+        });
+      }
+    },
+    async Transfer(from: string | null, to: string | null, tokenId: BigNumberish | null) {
+      // nothing
+    }
+  };
+  async function handleEvent(event: Event) {
```

Comment:
Since `handleEvent` is re-used, can you refactor into a common lib that each of `NftEvents`, `PoolEvents` and `PoolFactory` can use?